client/my-sites/checklist/controller.jsx

```diff
@@ -5,15 +5,39 @@
  */
 import React from 'react';
 import { get } from 'lodash';
+import page from 'page';
+import { isEnabled } from 'config';
 
 /**
  * Internal Dependencies
  */
 
 import ChecklistMain from './main';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import {
+	canCurrentUserUseCustomerHome,
+	canCurrentUserUseChecklistMenu,
+} from 'state/sites/selectors';
 
 export function show( context, next ) {
 	const displayMode = get( context, 'query.d' );
 	context.primary = <ChecklistMain displayMode={ displayMode } />;
 	next();
 }
+
+export function maybeRedirect( context, next ) {
+	const state = context.store.getState();
+	const siteId = getSelectedSiteId( state );
+	const slug = getSelectedSiteSlug( state );
+	if ( isEnabled( 'customer-home' ) && canCurrentUserUseCustomerHome( state, siteId ) ) {
+		page.redirect( `/home/${ slug }` );
```

Comment:
Since this returns to `page()`, would simply returning new URL as string work?

Reply:
Not sure if I understand your suggestion, but page.js handlers don't return anything and `next` doesn't take any parameters either (unlike express.js).

The only "API" is that you can modify the `context` and either call `next()` or don't.